modal/parallel_map.py

```diff
@@ -251,9 +251,7 @@ async def poll_outputs():
 
         assert len(received_outputs) == 0
 
-    async with aclosing(drain_input_generator()) as drainer, aclosing(pump_inputs()) as pump, aclosing(
-        poll_outputs()
-    ) as poller, aclosing(async_merge(drainer, pump, poller)) as streamer:
+    async with aclosing(async_merge(drain_input_generator(), pump_inputs(), poll_outputs())) as streamer:
```

Comment:
`async_merge` now closes its input generators so no need to wrap them in their own `aclosing` statement